File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.java
class Solution {
    public int maxProduct(int n) {
        if (n==0) return 0;
        int maxDigit = 0;
        int globalMax = 0;
        
        while (n > 0){
            int digit = n%10;
            globalMax = Math.max(globalMax, maxDigit*digit);
            maxDigit = Math.max(maxDigit,digit);
            
            n = n/10;
        }
        return globalMax;
    }
}

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        if n==0:
            return 0
        max_digit = 0
        global_max = 0    
        while n>0:
            digit = n % 10
            global_max = max(global_max, max_digit * digit)
            max_digit = max(max_digit, digit)
            n//=10
        return global_max    
